src/types/qwlayershellv1.h

```diff
@@ -3,75 +3,47 @@
 
 #pragma once
 
-#include <qwglobal.h>
-#include <qwconfig.h>
-#include <QObject>
+#include <qwobject.h>
 
-struct wlr_layer_surface_v1;
-struct wlr_layer_shell_v1;
-struct wl_resource;
-struct wlr_surface;
-
-using wlr_surface_iterator_func_t = void (*)(wlr_surface *surface, int sx, int sy, void *data);
+extern "C" {
+#include <math.h>
+#define namespace scope
+#define static
+#include <wlr/types/wlr_layer_shell_v1.h>
+#undef namespace
+#undef static
+}
 
 QW_BEGIN_NAMESPACE
 
-class QWDisplay;
-class QWSurface;
-class QWLayerSurfaceV1;
-class QWLayerShellV1Private;
-class QWXdgPopup;
-class QW_EXPORT QWLayerShellV1 : public QWWrapObject
+class QW_CLASS_OBJECT(layer_surface_v1)
 {
+    QW_OBJECT
     Q_OBJECT
-    QW_DECLARE_PRIVATE(QWLayerShellV1)
-public:
-    static QWLayerShellV1 *create(QWDisplay *display, uint32_t version);
-    inline wlr_layer_shell_v1 *handle() const {
-        return QWObject::handle<wlr_layer_shell_v1>();
-    }
-
-    static QWLayerShellV1 *get(wlr_layer_shell_v1 *handle);
-    static QWLayerShellV1 *from(wlr_layer_shell_v1 *handle);
 
-Q_SIGNALS:
-    void newSurface(QWLayerSurfaceV1 *surface);
+public:
+    QW_SIGNAL(new_popup, xdg_popup*)
 
-private:
-    QWLayerShellV1(wlr_layer_shell_v1 *handle, bool isOwner);
-    ~QWLayerShellV1() = default;
+public:
+    QW_FUNC_STATIC(layer_surface_v1, create)
+    QW_FUNC_STATIC(layer_surface_v1, try_from_wlr_surface)
+
+    QW_FUNC_MEMBER(layer_surface_v1, destroy)
```

Comment:
Is it has the wlr_layer_surface_v1_destroy function?

Reply:
`void wlr_layer_surface_v1_destroy(struct wlr_layer_surface_v1 *surface);`